File: csrc/26.cpp
//
// Created by linkinpark213 on 5/6/19.
//

#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int> &nums) {
        if (nums.size() == 0) return 0;
        if (nums.size() == 1) return 1;
        int value = nums[0];
        int count = 1;
        int i = 1;
        while (count < nums.size()) {
            if (nums[count] != value) {
                value = nums[count];
                count++;
            } else {
                nums.erase(nums.begin() + count, nums.begin() + count + 1);
            }
        }
        return count;
    }
};

int main() {
    Solution solution;
    int array[] = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
    vector<int> nums(array, array + sizeof(array) / sizeof(int));
    cout << solution.removeDuplicates(nums);
}

File: pysrc/26.py
from typing import List


class Solution:
    def printRemoveDuplicates(self, nums: List[int]) -> str:
        len = self.removeDuplicates(nums)
        ans = str(nums[:len])
        return ans

    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) <= 1:
            return len(nums)

        l, r = 0, 1
        val = nums[l]
        while r < len(nums):
            while r < len(nums) and val == nums[r]:
                r += 1
            if r < len(nums):
                val = nums[r]
                l += 1
                nums[l] = val

        return l + 1


if __name__ == '__main__':
    solution = Solution()
    print(solution.printRemoveDuplicates(nums=[1, 1, 2]))
    print(solution.printRemoveDuplicates(nums=[0, 0, 1, 1, 1, 2, 2, 3, 3, 4]))
